Runtime/Event/BehaviourEnabledObserver.cs

```diff
@@ -1,53 +1,136 @@
+        /// <summary>
+        /// The maximum amount of time to perform the <see cref="Behaviour.enabled"/> check before ending.
+        /// </summary>
+        [Serialized]
+        [field: DocumentedByXml]
+        public float MaximumRunTime { get; set; } = float.PositiveInfinity;
 
         /// <summary>
         /// The <see cref="Behaviour"/>s to observe.
         /// </summary>
-        [DocumentedByXml]
-        public List<Behaviour> behaviours = new List<Behaviour>();
+        [Serialized]
+        [field: DocumentedByXml]
+        public BehaviourObservableList Behaviours { get; set; }
+
         /// <summary>
-        /// Emitted when all <see cref="behaviours"/> are <see cref="Behaviour.isActiveAndEnabled"/>.
+        /// Emitted when all <see cref="Behaviours"/> are <see cref="Behaviour.isActiveAndEnabled"/>.
         /// </summary>
         [DocumentedByXml]
         public UnityEvent ActiveAndEnabled = new UnityEvent();
 
+        /// <summary>
+        /// A reference to the started routine.
+        /// </summary>
+        protected Coroutine behaviourCheckRoutine;
+        /// <summary>
+        /// Delays the <see cref="behaviourCheckRoutine"/> by <see cref="CheckDelay"/> seconds.
+        /// </summary>
+        protected WaitForSeconds checkDelayYield;
+
+        /// <summary>
+        /// Initiates the check of the <see cref="Behaviours"/> state if no existing check is already running.
+        /// </summary>
+        public virtual void BeginCheck()
+        {
+            if (behaviourCheckRoutine == null)
+            {
+                behaviourCheckRoutine = StartCoroutine(Check());
+            }
+        }
+
+        /// <summary>
+        /// Cancels any running check of the <see cref="Behaviours"/> state.
+        /// </summary>
+        public virtual void EndCheck()
+        {
+            if (behaviourCheckRoutine != null)
+            {
+                StopCoroutine(behaviourCheckRoutine);
+            }
+            behaviourCheckRoutine = null;
+        }
+
         protected virtual void OnEnable()
         {
-            InvokeRepeating(nameof(Check), MinimumRepeatRate, MinimumRepeatRate);
+            BeginCheck();
         }
 
         protected virtual void OnDisable()
         {
-            CancelInvoke(nameof(Check));
+            EndCheck();
+        }
+
+        /// <summary>
+        /// Checks to see if the <see cref="Behaviours"/> specified have been enabled in the scene.
+        /// </summary>
+        /// <returns>An Enumerator to manage the running of the Coroutine.</returns>
+        protected virtual IEnumerator Check()
+        {
+            float normalizedTime = 0;
+            checkDelayYield = new WaitForSeconds(CheckDelay);
+            while (normalizedTime <= 1f)
+            {
+                if (AreBehavioursEnabled())
+                {
+                    break;
+                }
+                normalizedTime += Time.deltaTime / MaximumRunTime;
```

Comment:
I don't understand this... Shouldn't the check just be like `float maximumTime = Time.time;` then `while (Time.time < maximumTime)`?

Reply:
you mean like

```
float endTime = Time.time + MaximumRunTime;
while (Time.time < endTime)
```